### ISSUE TITLE
Last Will (graceful disconnect) bug

### ISSUE THREAD

**Issue by Allineer** [wialon/gmqtt#59]
Sample:

```python
import asyncio

from gmqtt import Client, Message
from gmqtt.mqtt.constants import MQTTv311

import logging

STOP = asyncio.Event()


def on_connect(client, flags, rc, properties):
    print('CONNECTED')
    client.subscribe('test/#', qos=0)


def on_message(client, topic, payload, qos, properties):
    print('RECV MSG:', payload)


def on_disconnect(client, packet, exc=None):
    print('DISCONNECTED')


def on_subscribe(client, mid, qos):
    print('SUBSCRIBED')


async def main():
    client = Client("client-id", will_message=Message('test', "LOST", qos=1, retain=True))

    client.on_connect = on_connect
    client.on_message = on_message
    client.on_disconnect = on_disconnect
    client.on_subscribe = on_subscribe

    client.set_auth_credentials('username', 'password')
    await client.connect('127.0.0.1', 8883, keepalive=60, version=MQTTv311)

    client.publish('test', 'ONLINE', qos=1, retain=True)
    client.publish('test', 'OFFLINE', qos=1, retain=True)
    await client.disconnect() 

if __name__ == '__main__':
    asyncio.run(main())

```

Broker log:

```
test ONLINE
test OFFLINE
test LOST
```

Why LWT was sent?

**Comment by Lenka42** [comment]
Hi @Allineer 
I'm investigating your issue. Which broker do you use?

**Comment by Allineer** [comment]
@Lenka42 , 

```
Package: mosquitto
Version: 1.5.7-1
```

```
Operating System: Debian GNU/Linux 10 (buster)
            Kernel: Linux 4.9.0-8-amd64
      Architecture: x86-64
```

**Comment by Lenka42** [comment]
@Allineer I fixed bug in disconnect packet for MQTT 3.1.1, please check version 0.4.1

**Comment by Allineer** [comment]
It's working!

Спасибо, Елена!
